src/Servers/Kestrel/Core/test/MessageBodyTests.cs

```diff
@@ -778,8 +778,10 @@ public async Task CopyToAsyncThrowsOnTimeout()
 
                 // Add some input and read it to start PumpAsync
                 input.Add("a");
-                Assert.Equal(1, (await body.ReadAsync()).Buffer.Length);
+                var readResult = await body.ReadAsync();
+                Assert.Equal(1, readResult.Buffer.Length);
 
+                body.AdvanceTo(readResult.Buffer.End);
```

Comment:
What happens today if app middleware exits between a call to ReadAsync/AdvanceTo?

Reply:
I see ContentLengthReadAsyncPipeReaderBufferRequestBody does exactly that. Why did this test have to change?

Reply:
Not that it's wrong, but why did this test change?

Reply:
I need to learn to refresh lol.

Reply:
LOL